packages/zowe-explorer/src/Profiles.ts

```diff
@@ -839,12 +839,12 @@ export class Profiles extends ProfilesCache {
             await readConfigFromDisk();
             return newProfileName;
         } catch (error) {
-            await errorHandling(error.message);
+            await errorHandling(error, profileName, error.message);
             ZoweExplorerExtender.showZoweConfigError(error.message);
         }
     }
 
-    public async promptCredentials(sessionName: string, rePrompt?: boolean): Promise<string[]> {
+    public async promptCredentials(profile: string | zowe.imperative.IProfileLoaded, rePrompt?: boolean): Promise<string[]> {
```

Comment:
Hey @t1m0thyj why would we keep the option of string | IProfileLoaded?  Should we just make it IProfileLoaded and update any calls to it?

Reply:
I know this is only code in the zowe-explorer package.
But is it possible for extenders to call methods exported from the ZE's VSIX file (as a module)? (probably not recommended - if it's possible)

If that's the case, we may want to be careful with changing function signatures without maintaining the original implementation. 😋 

If we are not concerned about that, then we can  just replace the string and update our calls 😋

Reply:
we shouldn't have to worry about extenders trying to use ZE methods and should be able to adjust that code at any time unlike the ZE API package that is recommended for use to extend Zowe Explorer.

Reply:
I agree that since this method isn't in the ZE API package, we shouldn't need to be concerned about making breaking changes to its signature.

I kept the option of passing a string because we call `promptCredentials` that way in our `errorHandling` method:
https://github.com/zowe/vscode-extension-for-zowe/blob/6fa16aad1897aa1fa339cc16fe10662991b724ec/packages/zowe-explorer/src/utils/ProfilesUtils.ts#L104

Within the `errorHandling` method we don't have access to an `IProfileLoaded` object, just the node label.
I'm open to refactoring this if we want but it would add scope to the PR since the `errorHandling` method is called around 70 times across our codebase.